bpfd/src/command.rs

```diff
@@ -476,6 +404,28 @@ pub(crate) struct UprobeProgramInfo {
     pub(crate) namespace: Option<String>,
 }
 
+impl UprobeProgram {
+    pub(crate) fn new(
+        data: ProgramData,
+        fn_name: Option<String>,
+        offset: u64,
+        target: String,
+        retprobe: bool,
+        pid: Option<i32>,
+        namespace: Option<String>,
+    ) -> Self {
+        Self {
+            data,
+            fn_name,
+            offset,
+            target,
+            retprobe,
+            pid,
+            namespace,
+        }
+    }
+}
+
 impl Program {
     pub(crate) fn kind(&self) -> ProgramType {
```

Comment:
Should some of the accessor functions in `Program` get renamed to be more like those in `ProgramData` (which I liked). For example: `kind()` -> `get_program_type()` and `dispatcher_id()` -> `get_dispatcher_id()`?

Reply:
https://github.com/rust-lang/rfcs/blob/master/text/0344-conventions-galore.md#gettersetter-apis

The convention is `foo` and `set_foo`.
I agree though that the getter/setter should match the field name.
In cases where it doesn't we should fix that up also.

Reply:
Fixed up 👍